CQLEngine.js

```diff
@@ -0,0 +1,235 @@
+     * Fetches metadata from the CQL engine.
+     * @param {boolean} [force=false] - Whether to force fetching metadata.
+     * @returns {Promise<void>} - A Promise that resolves when metadata is fetched.
+     */
+    async fetch(force = false) {
+        if (this.baseURL) {
+            if (!this.metadata || force) {
+                try {
+                    let response = await axios.get(`${this.baseURL}/metadata`);
+                    if (response?.data) {
+                        this.metadata = response.data;
+                    }
+                } catch (e) {
+                    console.error(e);
+                }
+            }
+        }
+    }
+
+    /**
+     * Sets the API URL.
+     * @param {string} apiUrl - The API URL.
+     */
+    set apiUrl(apiUrl) {
+        this.info['apiUrl'] = apiUrl;
+    }
+
+    /**
+     * Gets the API URL.
+     * @returns {string|null} - The API URL.
+     */
+    get apiUrl() {
+        return this.info?.apiUrl ?? null;
+    }
+
+    /**
+     * Sets the CQL version.
+     * @param {string} version - The CQL version.
+     */
+    set cqlVersion(version) {
+        this.info['cqlVersion'] = version;
+    }
+
+    /**
+     * Gets the CQL version.
+     * @returns {string|null} - The CQL version.
+     */
+    get cqlVersion() {
+        return this.info?.cqlVersion ?? null;
+    }
+
+    /**
+     * Sets the CQL translator.
+     * @param {string} translator - The CQL translator.
+     */
+    set cqlTranslator(translator) {
+        this.info['cqlTranslator'] = translator;
+    }
+
+    /**
+     * Gets the CQL translator.
+     * @returns {string|null} - The CQL translator.
+     */
+    get cqlTranslator() {
+        return this.info?.cqlTranslator ?? null;
+    }
+
+    /**
+     * Sets the CQL translator version.
+     * @param {string} version - The CQL translator version.
+     */
+    set cqlTranslatorVersion(version) {
+        this.info['cqlTranslatorVersion'] = version;
+    }
+
+    /**
+     * Gets the CQL translator version.
+     * @returns {string|null} - The CQL translator version.
+     */
+    get cqlTranslatorVersion() {
+        return this.info?.cqlTranslatorVersion ?? null;
+    }
+
+    /**
+     * Sets the CQL engine.
+     * @param {string} engine - The CQL engine.
+     */
+    set cqlEngine(engine) {
+        this.info['cqlEngine'] = engine;
+    }
+
+    /**
+     * Gets the CQL engine.
+     * @returns {string|null} - The CQL engine.
+     */
+    get cqlEngine() {
+        return this.info?.cqlEngine ?? null;
+    }
+
+    /**
+     * Sets the CQL engine version.
+     * @param {string} version - The CQL engine version.
+     * @throws {Error} - If the version is not valid.
+     */
+    set cqlEngineVersion(version) {
+        if (version) {
+            if (CQLEngine.engineVersions.includes(version)) {
+                this.info['cqlEngineVersion'] = version;
+            } else {
+                throw new Error(`Not a valid CQL Engine version !!`);
```

Comment:
I'm not sure we want to limit this? Shouldn't we just allow an engine to report whatever it's version is?